cdap-ui/app/features/hydrator/controllers/tabs/status-ctrl.js

```diff
@@ -24,15 +24,22 @@ angular.module(PKG.name + '.feature.hydrator')
 
     $scope.programType = HydratorDetail.programType;
 
-    HydratorDetail.api.runs(params)
+    HydratorDetail.api.pollRuns(params)
       .$promise
       .then(function (runs) {
         $scope.stats.numRuns = runs.length;
         $scope.stats.lastRunTime = runs.length > 0 && runs[0].end ? runs[0].end - runs[0].start : 'N/A';
+
+        for (var i = 0; i < runs.length; i++) {
+          if (runs[i].status !== 'RUNNING') {
```

Comment:
A potential status is 'STARTING', which is not finished, but this code would assume it is finished, right?